dandi/support/bids/validator.py

```diff
@@ -50,23 +54,19 @@ def _get_paths(bids_paths):
         ".bidsignore",
         "dandiset.yaml",
     ]
-    # Inelegant hard-coded solution.
-    # Could be replaced by a maximum depth limit if BIDS root auto-detection is implemented.
-    treat_as_file_suffix = [".ngff"]
 
     path_list = []
     for bids_path in bids_paths:
         bids_path = os.path.abspath(os.path.expanduser(bids_path))
         if os.path.isfile(bids_path):
-            path_list.append(bids_path)
+            path_list.append(Path(bids_path).as_posix())
             continue
-        for root, dirs, file_names in os.walk(bids_path, topdown=False):
-            if any(root.endswith(i) for i in treat_as_file_suffix):
-                continue
-            if any(f"{i}/" in root for i in treat_as_file_suffix):
-                continue
-            if any(f"{i}\\" in root for i in treat_as_file_suffix):
-                continue
+        for root, dirs, file_names in os.walk(bids_path, topdown=True):
+            if root.endswith(tuple(pseudofile_suffixes)):
+                # Add the directory name to the validation paths list.
+                path_list.append(Path(root).as_posix() + "/")
+                # Do not index the contents of the directory.
+                dirs[:] = []
             # will break if BIDS ever puts meaningful data under `/.{dandi,datalad,git}*/`
             if any(exclude_subdir in root for exclude_subdir in exclude_subdirs):
```

Comment:
I suggest removing the `os.sep`s from `exclude_subdirs` and changing this line to:

```suggestion
            if os.path.basename(root) in exclude_subdirs:
```

Unlike the current code, this correctly handles things like `foo/.dandibar`.

Better yet, apply `exclude_subdirs` to `dirs` rather than `root` in order to actually avoid recursing into excluded subdirs.